editorconfig.go

```diff
@@ -98,7 +98,10 @@ func newDefinition(config Config) (*Definition, error) {
 		dir = filepath.Dir(dir)
 		ecFile := filepath.Join(dir, config.Name)
 		fp, err := os.Open(ecFile)
-		if os.IsNotExist(err) {
+		if err != nil {
+			if !os.IsNotExist(err) {
```

Comment:
@greut I'd merge these two `if` statements into one:

```go
if err != nil && !os.IsNotExist(err) {
```

Reply:
if the file doesn't exist, you want to _continue_. If something else occurs e.g. bad permission you want to fail as loudly as possible (which wasn't the case before that change).

Reply:
AFAIK there would be no change in the behavior of the code.

Reply:
the `Parse(nil)` won't make the runtime happy.